accounts/api/views.py

```diff
@@ -237,3 +241,43 @@ def signup(self, request):
             'user': UserSerializer(user).data
         }, status=status.HTTP_201_CREATED)
 
+
+class UserProfileViewSet(viewsets.GenericViewSet):
+    queryset = UserProfile.objects.all()
+    serializer_class = UserProfileSerializerForUpdate
+
+    def get_permissions(self):
+        if self.action == 'update':
+            return [IsAuthenticated(), IsObjectOwner()]
+        return [AllowAny()]
+
+    def list(self, request):
+        return Response({
+            'message': {
+                'Update profile': {
+                    'method': 'PUT',
+                    'url': '/api/profiles/{profile_id_pk}/',
+                },
+            }
+        }, status=status.HTTP_200_OK)
+
+    # <Wayne Shih> 22-Mar-2022
+    # URL:
+    # - PUT /api/profiles/{pk}/
+    # Typically use PUT to perform partial update.
+    @required_params(method='PUT', params=['nickname', 'avatar'], is_required_all=False)
+    def update(self, request, *args, **kwargs):
+        profile = self.get_object()
+        serializer = self.get_serializer(instance=profile, data=request.data)
+        if not serializer.is_valid():
+            return Response({
+                'success': False,
+                'message': 'Please check input.',
+                'errors': serializer.errors,
+            }, status=status.HTTP_400_BAD_REQUEST)
+
+        profile = serializer.save()
+        return Response(
+            UserProfileSerializerForUpdate(profile).data,
+            status=status.HTTP_200_OK,
+        )
```

Comment:
you might consider to use UpdateModelMixin

Reply:
I will consider the use case and if it is proper to use UpdateModelMixin, I will submit it in the next PR.

Reply:
![image](https://user-images.githubusercontent.com/65328145/160220703-69e8b821-30f6-4d99-b5ca-ac992f2f5325.png)
It turns out that using UpdateModelMixin would make some response messages inconsistent with other APIs.
I would like to stay with the current update() API instead of the one from UpdateModelMixin to keep response messages consistent with other APIs.

Reply:
Good pt!